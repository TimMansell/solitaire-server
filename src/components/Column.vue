<template>
  <div class="column">
    <Card
      v-for="(card, index) in cards"
      :key="index"
      :value="card.value"
      :suit="card.suit"
      :visible="card.visible" />
  </div>
</template>

<script>
import Card from '@/components/Card.vue';

export default {
  name: 'Column',
  components: {
    Card,
  },
  props: {
    cards: {
      type: Array,
      default: () => [],
    },
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.column {
  flex: 1;
  border: 1px solid white;
  padding: .5rem;
}
</style>
