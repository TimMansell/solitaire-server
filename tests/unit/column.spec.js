import { shallowMount } from '@vue/test-utils';
import Column from '@/components/Column.vue';

describe('Column.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Column);

    expect(wrapper).toMatchSnapshot();
  });
});
