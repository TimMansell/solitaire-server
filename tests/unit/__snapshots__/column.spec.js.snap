// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column.vue matches snapshot 1`] = `<div class="column"></div>`;
